### PR TITLE
Run rubocop on stdlib

### PR DESCRIPTION
**Pull request by iliabylich**
Last part!

### CODE REVIEW

File: stdlib/nodejs/require.rb [hunk 1]
```diff
@@ -16,15 +16,15 @@ def __prepare_require__(path)
     end
 
     name
-  rescue => e
-    raise [path, name, full_path].inspect+e.message
+  rescue StandardError => e
```
Comment (elia) [inline stdlib/nodejs/require.rb]: looked up in the style guide and didn't find it enforced, let alone a good reason to do that, can the rule be disabled if present?

File: stdlib/nodejs/rubygems.rb [hunk 2]
```diff
@@ -56,11 +56,9 @@ def curl url
...
 when 'install'
   name = ARGV.shift
-  if ARGV.include? '-v'
-    version = ARGV[ ARGV.index('-v')+1 ]
-  else
-    version = nil
-  end
+  version = if ARGV.include? '-v'
+              ARGV[ARGV.index('-v') + 1]
+            end
```
Comment (elia) [inline stdlib/nodejs/rubygems.rb]: For this code the author clearly wanted to highlight the fact that version can end up intentionally nil, I'm ok with the original version, another good one would be:

```rb
version = nil
version = ARGV[ARGV.index('-v') + 1] if ARGV.include? '-v'
```

_note 1: I didn't actually check who the author was (could be me of course)_
_note 2: I suspect this file could be removed entirely_ 😄

File: stdlib/opal-platform.rb [hunk 3]
```diff
@@ -3,13 +3,10 @@
...
-else # possibly browser, which is the primary target
-  OPAL_PLATFORM = nil
-end
+OPAL_PLATFORM = if nashorn
+                  'nashorn'
+                elsif node
+                  'nodejs'
+                elsif headless_chrome
+                  'headless-chrome'
+                end
```
Comment (elia) [inline stdlib/opal-platform.rb]: I find the original form much better, also the comment disappeared

Since I also don't like very much all the other changes of this kind (mostly `case/when`) can we ditch this rule altogether?

File: stdlib/ostruct.rb [hunk 4]
```diff
@@ -15,26 +17,33 @@ def []=(name, value)
...
-        raise ArgumentError.new "wrong number of arguments (0 for 1)"
+        raise ArgumentError, 'wrong number of arguments (0 for 1)'
       end
-      @table[new_ostruct_member(name[0 .. -2])] = args[0]
+      @table[new_ostruct_member(mid[0..-2])] = args[0]
+    elsif args.empty?
+      @table[mid.to_sym]
     else
-      @table[name.to_sym]
+      super
```
Comment (elia) [inline stdlib/ostruct.rb]: Is this an unrelated fix? If so maybe it's worth its own PR or going straight to master

File: stdlib/ostruct.rb [hunk 5]
```diff
@@ -15,26 +17,33 @@ def []=(name, value)
...
     end
   end
 
+  def respond_to_missing?(mid, include_private = false)
+    mname = mid.to_s.chomp('=').to_sym
+    @table&.key?(mname) || super
+  end
+
```
Comment (elia) [inline stdlib/ostruct.rb]: same as above

File: stdlib/promise.rb [hunk 6]
```diff
@@ -146,7 +146,7 @@ def exception?
   end
 
   def realized?
-    !!@realized
+    @realized != false
```
Comment (elia) [inline stdlib/promise.rb]: This is not equivalent, did you check if the new code is safe?

File: stdlib/nodejs/irb.rb [hunk 7]
```diff
@@ -2,44 +2,39 @@
...
+                               prompt_interrupted = false
+                               $repl.prompt = prompt.call(self)
+                               begin
+                                 result = `OpalNode.run(cmd, filename)`
+                                 result = nil if `#{result} == nil`
+                                 callback.call('=> ' + result.inspect)
+                               rescue StandardError => e
+                                 callback.call(e.backtrace.join("\n"))
+                               end
+                             }
```
[the remark this answers is not in the document]
Comment (iliabylich) [reply stdlib/nodejs/irb.rb]: Agree, I'll fix it

File: stdlib/nodejs/rubygems.rb (hunk 2 above)
Comment (iliabylich) [reply stdlib/nodejs/rubygems.rb]: Yeah, it contains some totally weird code (there was an interpolation in the single-quoted string and it invokes `system` in a couple of places)

File: stdlib/ostruct.rb (hunk 4 above)
Comment (iliabylich) [reply stdlib/ostruct.rb]: That's what MRI does. Would you like me to revert it, add to filters for now and send it in a separate PR?

File: stdlib/promise.rb (hunk 6 above)
Comment (iliabylich) [reply stdlib/promise.rb]: Yes, `@realized` can be `false` or a symbol, also all specs pass (there are some in `spec/opal/stdlib/promise/*.rb` I guess)

File: stdlib/ostruct.rb (hunk 4 above)
Comment (elia) [reply stdlib/ostruct.rb]: @iliabylich commit straight to master, the diff in this PR should become clear after that

**Comment by iliabylich** [comment]
@elia Ping

File: stdlib/nodejs/irb.rb [hunk 8]
```diff
@@ -2,44 +2,42 @@
...
+          prompt_interrupted = false
+          $repl.prompt = prompt.call(self)
+          begin
+            result = `OpalNode.run(cmd, filename)`
+            result = nil if `#{result} == nil`
+            callback.call('=> ' + result.inspect)
+          rescue => e
+            callback.call(e.backtrace.join("\n"))
+          end
+        },
```
Comment (elia) [inline stdlib/nodejs/irb.rb]: It's ok anyway but I think the body of this proc should be just one indentation level more than `eval:`

File: stdlib/opal-platform.rb [hunk 9]
```diff
@@ -3,13 +3,11 @@
...
-  OPAL_PLATFORM = nil
-end
+OPAL_PLATFORM = if nashorn
+                  'nashorn'
+                elsif node
+                  'nodejs'
+                elsif headless_chrome
+                  'headless-chrome'
+                else # possibly browser, which is the primary target
+                end
```
Comment (elia) [inline stdlib/opal-platform.rb]: still prefer the earlier version with its regular repetition, also the missing nil bothers me

File: stdlib/pathname.rb [hunk 10]
```diff
@@ -87,23 +87,23 @@ def plus(path1, path2) # -> path # :nodoc:
...
-    while r2 = chop_basename(prefix2)
+    while (r2 = chop_basename(prefix2))
       prefix2, basename2 = r2
       index_list2.unshift prefix2.length
       basename_list2.unshift basename2
     end
     return path2 if prefix2 != ''
     prefix1 = path1
-    while true
+    loop do
```
Comment (elia) [inline stdlib/pathname.rb]: I'll guess 🤖 👮 has a good reason for doing this but in opal while true looks faster:

```
Warming up --------------------------------------
          while true   407.000  i/100ms
                loop   184.000  i/100ms
Calculating -------------------------------------
          while true      7.088k (± 2.6%) i/s -     35.409k in   4.999414s
                loop      1.970k (± 2.6%) i/s -      9.936k in   5.046192s

Comparison:
          while true:     7087.7 i/s
                loop:     1970.5 i/s - 3.60x  slower
```

code:

```rb
Benchmark.ips do |x|
  x.report('while true') do
    n = 0
    while true
      n += 1
      break if n == 10000
    end
  end

  x.report('loop') do
    n = 0
    loop do
      n += 1
      break if n == 10000
    end
  end

  x.compare!
end
```

**Comment by iliabylich** [comment]
@elia I've rebased this branch on the master, the build is green. Any comments?

**Comment by iliabylich** [comment]
@elia Thanks for patience!
